### PR TITLE
fixes event-biomass trimming

### LINKED ISSUE #15859: Tomahawks don't work on biomass
(WEB REPORT BY: panthid REMOTE: 172.93.109.202:7777)
# Revision
089fa1318a411f50e6db33a58c0eb620d7e4eea7

# Description
Tomahawks don't work on biomass (space barf) despite being a child of hatchets. There's probably more sharp weapons that it doesn't check for too.
# Steps to Reproduce
Try to clear biomass with a tomahawk.
# What you Expected
Tomahawks to clear biomass.
# What Actually Happened
My tomahawk didn't clear biomass.

### PR DESCRIPTION
**Pull request by MadmanMartian**
Now makes use of sharpness flags and heat generation rather than the abortion it was previously

closes #15859

### CODE REVIEW

**Comment by ghost** [comment]
Does this generate one message per tile

**Comment by MadmanMartian** [comment]
Yes

**Comment by N3X15** [comment]
That's going to spam like a motherfucker with scythes and whatnot.

File: code/game/gamemodes/events/biomass.dm [hunk 1]
```diff
@@ -23,45 +23,32 @@
...
+			"<span class = 'notice'>You use \the [src]'s hot edge to burn through \the [src].</span>")
 			qdel(src)
-		if(/obj/item/weapon/melee/energy)
+			return
+	var/weapon_temp = W.is_hot()
+	if(weapon_temp >= AUTOIGNITION_WOOD)//Yes it's not technically wood, but fibrous chitin's pretty close when held above a flame
+		var/coeff = 1*weapon_temp/AUTOIGNITION_WOOD //The hotter it is, the less time it takes
+		if(do_after(user, src, (rand(30,60)/coeff)))
+			user.visible_message("<span class = 'warning'>\The [user] burns away \the [src] with \the [W].</span>",\
+			"<span class = 'notice'>You use \the [src] to burn away \the [src].</span>")
```
Comment (Exxion) [inline code/game/gamemodes/events/biomass.dm]: [Honk]

File: code/game/gamemodes/events/biomass.dm [hunk 2]
```diff
@@ -23,45 +23,32 @@
...
+		if(do_after(user, src, rand(10,30)))
+			user.visible_message("<span class = 'warning'>\The [user] chops through \the [src] with \the [W].</span>",\
+			"<span class = 'notice'>You saw through \the [src].</span>")
 			qdel(src)
-		if(/obj/item/weapon/hatchet)
+			return
+	if(W.sharpness_flags & HOT_EDGE)
+		if(do_after(user, src, rand(5,15))) //Guaranteed, rarer sharpness flag so less time taken
+			user.visible_message("<span class = 'warning'>\The [user] sears through \the [src] with \the [W].</span>",\
+			"<span class = 'notice'>You use \the [src]'s hot edge to burn through \the [src].</span>")
```
Comment (Exxion) [inline code/game/gamemodes/events/biomass.dm]: >Using it as a weapon against itself

I like it.

Comment (MadmanMartian) [reply code/game/gamemodes/events/biomass.dm]: whoops

**Pull request merged**